### PR TITLE
refactor(dart/transform): Run `ReflectionRemover` after codegen phases

### PR DESCRIPTION
**Pull request by kegluneq**
Move the `ReflectionRemover` phase to execute after the codegen phases.

### CODE REVIEW

File: modules_dart/transform/lib/src/transform/common/ast_tester.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import 'package:analyzer/src/generated/ast.dart';
+import 'package:angular2/src/transform/common/names.dart';
+
+const BOOTSTRAP_STATIC_URI = 'package:angular2/bootstrap_static.dart';
+const BOOTSTRAP_URI = 'package:angular2/bootstrap.dart';
+const REFLECTION_CAPABILITIES_URI =
+    'package:angular2/src/core/reflection/reflection_capabilities.dart';
+
+/// An object that checks for {@link ReflectionCapabilities} syntactically.
```
Comment (yjbanov) [inline modules_dart/transform/lib/src/transform/common/ast_tester.dart]: `s/An object that//`

File: modules_dart/transform/lib/src/transform/common/ast_tester.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+class AstTester {
```
Comment (yjbanov) [inline modules_dart/transform/lib/src/transform/common/ast_tester.dart]: Would it be more consistent with `AnnotationMatcher` if we called it `AstMatcher`?

File: modules_dart/transform/lib/src/transform/common/ast_tester.dart (hunk 1 above)
Comment (kegluneq) [inline modules_dart/transform/lib/src/transform/common/ast_tester.dart]: Done

File: modules_dart/transform/lib/src/transform/common/ast_tester.dart (hunk 2 above)
Comment (kegluneq) [inline modules_dart/transform/lib/src/transform/common/ast_tester.dart]: What do you think of `MirrorMatcher`?

Comment (yjbanov) [inline modules_dart/transform/lib/src/transform/common/ast_tester.dart]: :thumbsup:

**Comment by kegluneq** [comment]
Committed as 6b40293c0a8931cc0c2cb0a28051d490f583ae51

**Pull request closed without merging**